test/functional/config/utils.go

```diff
@@ -163,97 +174,100 @@ func (u *TestUtils) runCmd(cmdline string) (string, error) {
 	return string(out), nil
 }
 
-func (u *TestUtils) AwaitIssuerReady(names ...string) error {
-	return u.AwaitState("issuer", "Ready", names...)
+func (u *TestUtils) WaitUntilIssuerReady(ctx context.Context, names ...string) error {
+	return u.WaitUntilIssuerState(ctx, "Ready", names...)
 }
 
-func (u *TestUtils) AwaitIssuerDeleted(names ...string) error {
-	return u.AwaitState("issuer", STATE_DELETED, names...)
+func (u *TestUtils) WaitUntilIssuerDeleted(ctx context.Context, names ...string) error {
+	return u.WaitUntilIssuerState(ctx, STATE_DELETED, names...)
 }
 
-func (u *TestUtils) AwaitCertReady(names ...string) error {
-	return u.AwaitState("cert", "Ready", names...)
+func (u *TestUtils) WaitUntilCertReady(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Ready", names...)
 }
 
-func (u *TestUtils) AwaitCertError(names ...string) error {
-	return u.AwaitState("cert", "Error", names...)
+func (u *TestUtils) WaitUntilCertError(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Error", names...)
 }
 
-func (u *TestUtils) AwaitCertDeleted(names ...string) error {
-	return u.AwaitState("cert", STATE_DELETED, names...)
+func (u *TestUtils) WaitUntilCertDeleted(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, STATE_DELETED, names...)
 }
 
-func (u *TestUtils) AwaitCertRevoked(names ...string) error {
-	return u.AwaitState("cert", "Revoked", names...)
+func (u *TestUtils) WaitUntilCertRevoked(ctx context.Context, names ...string) error {
+	return u.WaitUntilCertState(ctx, "Revoked", names...)
 }
 
-func (u *TestUtils) AwaitCertRevocationApplied(name string) error {
-	return u.AwaitState("certrevoke", "Applied", name)
+func (u *TestUtils) WaitUntilCertRevocationApplied(ctx context.Context, name string) error {
+	return u.WaitUntilCertificateRevocationState(ctx, "Applied", name)
 }
 
-func (u *TestUtils) AwaitState(resourceName, expectedState string, names ...string) error {
-	msg := fmt.Sprintf("%s not %s: %v", resourceName, expectedState, names)
-	return u.Await(msg, func() (bool, error) {
-		output, err := u.runKubeCtl("get " + resourceName + " \"-o=jsonpath={range .items[*]}{.metadata.name}={.status.state}{'\\n'}{end}\"")
-		if err != nil {
-			return false, err
+// WaitUntilCertState takes names of Certificates and waits for them to get ready with a timeout of 15 seconds.
```

Comment:
The timeout is determined by `u.AwaitTimeout`, right? I suggest removing the timeout length from the comment as they're already out of sync and it's easy to miss when changing the timeout value.
``` suggestion
// WaitUntilCertState takes the names of Certificates and waits for them to get ready with a timeout.
```